hibernate-core/src/test/java/org/hibernate/orm/test/query/TemporalParameterPlusDurationTest.java

```diff
@@ -0,0 +1,35 @@
+package org.hibernate.orm.test.query;
+
+import java.time.Instant;
+
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.JiraKey;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.junit.jupiter.api.Test;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+
+@DomainModel(annotatedClasses = TemporalParameterPlusDurationTest.SimpleEntity.class)
+@SessionFactory
+@JiraKey("HHH-18201")
+public class TemporalParameterPlusDurationTest {
+
+	@Test
+	void temporalParameterPlusDuration(SessionFactoryScope scope) {
+		scope.inSession( session -> {
+			session.createQuery( "from SimpleEntity where inst > :i + 1 second", SimpleEntity.class )
+					.setParameter( "i", Instant.now() )
+					.getResultCount();
+		} );
+	}
```

Comment:
Would you please also add a test for the contrary case where the parameter is a duration? 🙏

Reply:
I will, but the problem is that tests are failing for several databases :-(